src/criteria_assessment/site_identification.jl

```diff
@@ -0,0 +1,214 @@
+"""Methods for identifying potential deployment locations."""
+
+using FLoops, ThreadsX
+
+"""
+    proportion_suitable(subsection::BitMatrix, window::Tuple=(-4,5))::Matrix{Int16}
+
+Calculate the the proportion of the subsection that is suitable for deployments.
+Subsection is the surrounding a rough hectare area centred on each cell of a raster marked
+as being suitable according to user-selected criteria.
+
+# Arguments
+- `x` : Matrix of boolean pixels after filtering with user criteria.
+- `window` : Window size to assess. Default window (-4,5) assesses a square hectare around each target pixel where the resolution of pixels is 10m.
+"""
+function proportion_suitable(x::BitMatrix; window::Tuple=(-4, 5))::Matrix{Int16}
+    x′ = zeros(Int16, size(x))
+
+    @floop for row_col in ThreadsX.findall(x)
+        (row, col) = Tuple(row_col)
+        x_left = max(col + window[1], 1)
```

Comment:
could extract window tuple and save result of size calls as consts. Might improve performance a bit and would improve readability.

Reply:
Hang on, this is using the wrong offsets.

Reply:
Nevermind, I think it's good enough, just not what I initially expected.